internal/controllers/rest_controller.go

```diff
@@ -0,0 +1,62 @@
+package controllers
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/handlers"
+	"github.com/gorilla/mux"
+	"github.com/tevino/abool"
+	"github.com/ydataai/authentication-service/internal/clients"
+	"github.com/ydataai/authentication-service/internal/server"
+	"github.com/ydataai/go-core/pkg/common/logging"
+)
+
+// RESTController defines rest controller
+type RESTController struct {
+	serverConfig *server.Configuration
+	oidcConfig   *clients.OIDCConfiguration
+	logger       logging.Logger
+}
+
+// NewRESTController initializes rest controller
+func NewRESTController(logger logging.Logger,
+	sc *server.Configuration,
+	oc *clients.OIDCConfiguration) RESTController {
+	return RESTController{
+		serverConfig: sc,
+		oidcConfig:   oc,
+		logger:       logger,
+	}
+}
+
+// Boot ...
+func (r RESTController) Boot(s *server.Server) {
+	// Start readiness probe immediately
+	r.logger.Infof("Starting Readiness probe at %v", r.serverConfig.ReadinessProbePort)
+	isReady := abool.New()
+	go func() {
+		r.logger.Fatal(http.ListenAndServe(fmt.Sprintf(":%d", r.serverConfig.ReadinessProbePort), server.Readiness(isReady)))
+	}()
```

Comment:
🤔  this doesn't seem responsibility for the controller, even more for this stage of the boot